doc/api/deprecations.md

```diff
@@ -2617,6 +2617,39 @@ async function openAndClose() {
 }
 ```
 
+<a id="DEP0XXX"></a>
+### DEP0XXX: `module.parent`
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/32217
+    description: Documentation-only deprecation.
+-->
+
+Type: Documentation-only (supports [`--pending-deprecation`][])
+
+A CommonJS module can access the first module that required it using
+`module.parent`. This feature is deprecated because it does not work
+consistently in the presence of ECMAScript modules and because it gives an
+inaccurate representation of the CommonJS module graph.
+
+Some modules use it to check if they are the entry point of the current process.
+Instead, it is recommended to compare `require.main` and `module`:
+
+```js
+if (require.main === module) {
+  // CLI code runs here
```

Comment:
Given that nothing in the text mentions CLI's, this comment is a bit confusing.

Reply:
```suggestion
  // Code section that will run only if current file is the entry point.
```